### DISCUSSION TITLE
Unable to pop modal and navigate in the same function

### DISCUSSION THREAD

**Discussion by BruceTheBrick** [PrismLibrary/Prism#2719] [Xamarin.Forms] [answered]
Xamarin.Forms - 5.0.0.2478
Prism.DryIoc.Forms - 8.1.97

I've been making a little app for myself to help me get more familiar with both Xamarin.Forms and Prism. 
Currently I have an issue where I can't seem to pop a modal and navigate from the same function. 
My function is implemented like this:

``` C#
public async Task NavigateToFlowAsync(){
   await NavigationService.GoBackAsync(true);
   await NavigationService.NavigateAsync($"{nameof(NavigationPage)}/{nameof(NewPage)}");
}
```

This function is implemented and called from a page that is in the modal that's being popped. The idea is that this page will be the last in a flow that occurs within a modal, and the user can make a selection on this page that should pop the modal and then navigate to an arbitrary ```NewPage```. 

The current behavior is that the modal _is_ successfully popped, but the subsequent navigation doesn't happen. I've tried looking at the ```INavigationResult``` objects returned by these calls, and in both cases they are successful and have no exceptions. 

I'd greatly appreciate any help with this, and also welcome any feedback in regards to formulating my question. 
Thank you!

**Comment by brianlagunas** [comment, marked as answer]
You can't navigate from a page you just removed from the navigation stack.

**Reply by BruceTheBrick** [reply]
Makes a lot of sense. Thought that may have been the issue, but I was really hoping it wasnt!

Thanks for the help @brianlagunas. Really appreciate it!
